### PR TITLE
DAOS-16477 pool: Rename Suspect state to Dead

### PR DESCRIPTION
**Pull request by mjmac**
Change the name to more closely reflect the underlying
SWIM status, and reduce user confusion. An engine that
has been marked DEAD by SWIM cannot participate in pool
services, and has most likely already SIGKILL-ed itself.

Features: pool
Required-githooks: true

Signed-off-by: Michael MacDonald <mjmac@google.com>

### CODE REVIEW

File: src/include/daos_pool.h
Comment (liw) [inline src/include/daos_pool.h:165]: [Nit; not requesting changes] Maybe this degree of inaccuracy is acceptable, but I just want to clarify that these "dead" engines are actually the set of pool engines who are not "alive". In other words, they may also include engines absent from the SWIM membership.

**Review by liw — APPROVED** [review]

**Review by daltonbohning — COMMENTED** [review]
ftest LGTM

File: src/include/daos_pool.h
Comment (tanabarr) [reply src/include/daos_pool.h:165]: if that is the case then maybe suspect is a more accurate term?

**Review by tanabarr — APPROVED** [review]

**Comment by mjmac** [comment]
Test failure appears to be https://daosio.atlassian.net/issues/DAOS-16829, which was introduced with the PR that added the suspect state. I could re-run without `Features: pool`, but probably best to wait until the fix is merged and then re-run after merging.

File: src/include/daos_pool.h
Comment (mjmac) [reply src/include/daos_pool.h:165]: @liw: When would we have engines in the pool map that are absent from the SWIM membership, but aren't dead?

Comment (liw) [reply src/include/daos_pool.h:165]: When the MS has excluded the engines from the system (and the new system map has been distributed out), but the PS hasn't excluded the engines from the pool. In this state, from the pool's point of view, these engines are not strictly SWIM-DEAD, just unavailable---loosely speaking, "dead".

Hmm, personally I don't think "suspect" is more accurate. :)

**Comment by mjmac** [comment]
I've merged master and re-pushed with the `Allow-unstable-test: true` pragma set.

**Comment by mjmac** [comment]
Two known failures:
  * [DAOS-16874](https://daosio.atlassian.net/browse/DAOS-16874)
  * [DAOS-16829](https://daosio.atlassian.net/browse/DAOS-16829)

**Pull request merged**
